src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcher.java

```diff
@@ -19,480 +18,438 @@
-	 * 
-	 * @param region                   task region
-	 * @param initDate                 initial date
-	 * @param endDate                  end date
-	 * @param inputdownloadingPhaseTag inputdownloading phase tag
-	 * @param preprocessingPhaseTag    preprocessing phase tag
-	 * @param processingPhaseTag       processing phase tag
-	 * @return processed tasks list
-	 */
-	private List<SapsImage> getProcessedTasksInCatalog(String region, Date initDate, Date endDate,
-			String inputdownloadingPhaseTag, String preprocessingPhaseTag, String processingPhaseTag) {
-		return CatalogUtils.getProcessedTasks(imageStore, region, initDate, endDate, inputdownloadingPhaseTag,
-				preprocessingPhaseTag, processingPhaseTag,
-				"gets all processed tasks with region [" + region + "], inputdownloading tag ["
-						+ inputdownloadingPhaseTag + "], preprocessing tag [" + preprocessingPhaseTag
-						+ "], processing tag [" + processingPhaseTag + "] beetwen " + initDate + " and " + endDate);
-	}
-
-	/**
-	 * This function search all processed tasks from area (between latitude and
-	 * longitude coordinates) between initial date and end date with
-	 * inputdownloading, preprocessing and processing tags.
-	 * 
-	 * @param lowerLeftLatitude        lower left latitude (coordinate)
-	 * @param lowerLeftLongitude       lower left longitude (coordinate)
-	 * @param upperRightLatitude       upper right latitude (coordinate)
-	 * @param upperRightLongitude      upper right longitude (coordinate)
-	 * @param initDate                 initial date
-	 * @param endDate                  end date
-	 * @param inputdownloadingPhaseTag inputdownloading phase tag
-	 * @param preprocessingPhaseTag    preprocessing phase tag
-	 * @param processingPhaseTag       processing phase tag
-	 * @return processed tasks list following description
-	 */
-	public List<SapsImage> searchProcessedTasks(String lowerLeftLatitude, String lowerLeftLongitude,
-			String upperRightLatitude, String upperRightLongitude, Date initDate, Date endDate,
-			String inputdownloadingPhaseTag, String preprocessingPhaseTag, String processingPhaseTag) {
-
-		List<SapsImage> filteredTasks = new ArrayList<>();
-		Set<String> regions = regionsFromArea(lowerLeftLatitude, lowerLeftLongitude, upperRightLatitude,
-				upperRightLongitude);
-
-		for (String region : regions) {
-			List<SapsImage> tasksInCurrentRegion = getProcessedTasksInCatalog(region, initDate, endDate,
-					inputdownloadingPhaseTag, preprocessingPhaseTag, processingPhaseTag);
-			filteredTasks.addAll(tasksInCurrentRegion);
-		}
-		return filteredTasks;
-	}
+
+    private Catalog catalog;
+
+    private static final Logger LOGGER = Logger.getLogger(SubmissionDispatcher.class);
+
+    public SubmissionDispatcher(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
+    public SubmissionDispatcher(Properties properties) throws SQLException {
+        this.catalog = new JDBCCatalog(properties);
+    }
+
+    /**
```

Comment:
a reminder: there is no need to doc methods like this `getCatalog`. It is quite obvious. (since we doc it, fine. keep the doc)

Reply:
Remove it.